tests/test_foundation/test_base.py

```diff
@@ -52,3 +52,43 @@ def test_basic_attributes(self):
         self.assertIsNot(message1.b.c, message2.b.c)
         self.assertIsNot(message1.b.c.c1, message2.b.c.c1)
         self.assertIsNot(message1.b.c.c2, message2.b.c.c2)
+
+class TestGenericType(unittest.TestCase):
+
+    def setUp(self):
+        """Basic Test Setup."""
+        class MyGenericType(base.GenericType):
```

Comment:
Why you defined a new class here ? Just use the UBInt32 (and similar) already defined to test. Makes sense to you ?